nodes/list_struct/shuffle.py

```diff
@@ -55,30 +55,41 @@ def update(self):
             changable_sockets(self, inputsocketname, outputsocketname)
 
     def process(self):
-        if self.outputs['data'].is_linked:
+        if self.outputs[0].is_linked and self.inputs[0].is_linked:
 
             seed = self.inputs['seed'].sv_get()[0][0]
 
             random.seed(seed)
+            np_random.seed(seed)
             data = self.inputs['data'].sv_get()
             output = self.shuffle(data, self.level)
             self.outputs['data'].sv_set(output)
 
     def shuffle(self, lst, level):
         level -= 1
+        lst_type = type(lst)
         if level:
+            if level == 1 and lst_type == ndarray:
+                out = np.array(lst)
+                for row in out:
+                    np_random.shuffle(row)
+                return out
             out = []
             for l in lst:
                 out.append(self.shuffle(l, level))
             return out
-        elif type(lst) in [type([])]:
+        elif lst_type == list:
             l = lst.copy()
```

Comment:
Yes .... it is called 4 times in that function so i thought i was not a bad idea to calling it once.... maybe is a poor optimisation but it is a optimisation

Reply:
Some downsides of yo solution:
- Readability is worse. 
- It can bring troubles in future because `type() is` is not the same as `isinstace`. 
- There is no optimization here. This does not make it faster proportionally to amount of input data of the node. 
- `Is` should be used anyway.

Reply:
ok, changed 😄

Reply:
Another solution could be:
```Python
input_is_ndarray = isinstance(lst, np.ndarray)
if input_is_ndarray:
    ...
```